zcash_client_sqlite/src/testing/pool.rs

```diff
@@ -318,150 +314,252 @@ pub(crate) fn send_multi_step_proposed_transfer<T: ShieldedPoolTester>() {
     let account = st.test_account().cloned().unwrap();
     let dfvk = T::test_account_fvk(&st);
 
-    // Add funds to the wallet in a single note
+    let add_funds = |st: &mut TestState<_>, value| {
+        let (h, _, _) = st.generate_next_block(&dfvk, AddressType::DefaultExternal, value);
+        st.scan_cached_blocks(h, 1);
+
+        assert_eq!(
+            block_max_scanned(&st.wallet().conn, &st.wallet().params)
+                .unwrap()
+                .unwrap()
+                .block_height(),
+            h
+        );
+        assert_eq!(st.get_spendable_balance(account.account_id(), 1), value);
+    };
+
     let value = NonNegativeAmount::const_from_u64(100000);
-    let (h, _, _) = st.generate_next_block(&dfvk, AddressType::DefaultExternal, value);
-    st.scan_cached_blocks(h, 1);
+    let transfer_amount = NonNegativeAmount::const_from_u64(50000);
+
+    let run_test = |st: &mut TestState<_>, expected_index| {
+        // Add funds to the wallet.
+        add_funds(st, value);
+
+        let expected_step0_fee = (zip317::MARGINAL_FEE * 3).unwrap();
+        let expected_step1_fee = zip317::MINIMUM_FEE;
+        let expected_ephemeral = (transfer_amount + expected_step1_fee).unwrap();
+        let expected_step0_change =
+            (value - expected_ephemeral - expected_step0_fee).expect("sufficient funds");
+        assert!(expected_step0_change.is_positive());
```

Comment:
Instead of this assertion, convert the returned value to `NonNegativeAmount` and unwrap the error; this is a check on the implementation of the test setup, not a test assertion and it's confusing to use `assert!` for the former.

Reply:
It already is a `NonNegativeAmount`, and we already do unwrap (`.expect("sufficient funds")`). This is checking that it is non-zero, because zero is a corner case for the change handling that isn't what we want to be testing here.